kfs-core/src/main/java/edu/arizona/kfs/gl/businessobject/EntryBo.java

```diff
@@ -0,0 +1,397 @@
+
+        for (AccountingLine accountingLine : accountingLineList) {
+            String documentNumber = accountingLine.getDocumentNumber();
+            DocumentRouteHeaderValue docHeader = getBusinessObjectService().findBySinglePrimaryKey(DocumentRouteHeaderValue.class, documentNumber);
+            if (docHeader != null) {
+                String docTypeName = docHeader.getDocumentType().getName();
+                if (docTypeName.equals(KFSConstants.FinancialDocumentTypeCodes.GENERAL_ERROR_CORRECTION)) {
+                    return documentNumber;
+                }
+            }
+        }
+        return KFSConstants.EMPTY_STRING;
+    }
+
+    private String generateEntryId() {
+        StringBuilder retval = new StringBuilder();
+        retval.append(getUniversityFiscalYear());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getChartOfAccountsCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getAccountNumber());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getSubAccountNumber());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialObjectCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialSubObjectCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialBalanceTypeCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialObjectTypeCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getUniversityFiscalPeriodCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialDocumentTypeCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getFinancialSystemOriginationCode());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getDocumentNumber());
+        retval.append(SecConstants.SecurityValueSpecialCharacters.MULTI_VALUE_SEPERATION_CHARACTER);
+        retval.append(getTransactionLedgerEntrySequenceNumber());
+        return retval.toString();
+    }
+
+    /**
+     * Generates a Map of fieldValues for a Primary Key lookup based on the given entryId.
+     *
+     * @param entryId
+     * @return
+     */
+    public static Entry getEntry(String entryId) {

```

Comment:
That's not quite the point of this method. This method is to essentially retrieve the full Entry object including helper objects. It turns out that if a retrieved object contains helper objects, the BOService retrieves each of these one at a time in a loop anyway, so there's no significant difference. It'll end up being a "6 in one hand, half-dozen in the other" situation.